src/Compilers/Core/Portable/DiagnosticAnalyzer/CompilationWithAnalyzers.cs

```diff
@@ -885,7 +885,7 @@ private void GenerateCompilationEvents(AnalysisScope analysisScope, Cancellation
             else if (!analysisScope.IsSyntacticSingleFileAnalysis)
             {
                 var mappedModel = _compilation.GetSemanticModel(analysisScope.FilterFileOpt!.Value.SourceTree!);
-                _ = mappedModel.GetDiagnostics(cancellationToken: cancellationToken);
+                _ = mappedModel.GetDiagnostics(analysisScope.FilterSpanOpt, cancellationToken);
```

Comment:
i agree with sharwell, let's doc this thoroughly.    all the info in this pr woudl be valuable to have written up in complete form in this code to help in the future.